server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,674 @@
+                    if (submittedSmall){
+                        // if it is already valid, return
+                        if (submittedSmall.verified){
+                            return res.send(400, 'Small group attendance already recorded: ' + submittedSmall.verified);
+                        }
+                        else{
+                            submittedSmall.verified = true;
+                            submittedSmall.save();
+                            return res.send(200, {'type':'Small group attendance', 'unverified': needsVerification});
+
+                        }
+                    }
+                    // if not, create the attendance object
+                    Attendance.create({
+                        classYear: classYear._id,
+                        user: user._id,
+
+                        date: isoDateToTime(new Date()),
+                        datetime: new Date(),
+
+                        bonusDay: false,
+                        smallgroup: true,
+
+                        verified: !needsVerification,
+                        code: '',
+
+                    });
+                    return res.send(200, {'type':'Small group attendance', 'unverified': needsVerification});
+
+                }
+                // Small group & bonus day
+                else if (small && bonus){
+                    // Check if the user already submitted a small group & bonus attendance
+                    if (submittedSmallBonus){
+                        // if it is already valid, return
+                        if (submittedSmallBonus.verified){
+                            return res.send(400, 'Small group bonus attendance already recorded: ' + submittedSmallBonus.verified);
+                        }
+                        else{
+                            submittedSmallBonus.verified = true;
+                            submittedSmallBonus.save();
+                            return res.send(200, {'type':'Small group bonus attendance', 'unverified': needsVerification});
+
+                        }
+
+                    }
+                    // if not, create the attendance object
+                    Attendance.create({
+                        classYear: classYear._id,
+                        user: user._id,
+
+                        date: isoDateToTime(new Date()),
+                        datetime: new Date(),
+
+                        bonusDay: true,
+                        smallgroup: true,
+
+                        verified: !needsVerification,
+                        code: '',
+
+                    });
+                    return res.send(200, {'type':'Small group bonus attendance', 'unverified': needsVerification});
+
+                }
+            });
+        }
+    });
+
+};
+
+// Get data for submitting attendance, then pass it to setAttendance
+var setAttendanceInfo = function(req, res, small, bonus){
```

Comment:
why can't setAttendance handle this logic? Why do we need 5 other functions just to call setAttendance?